### PR TITLE
Simplifies the implementation of CommonUtility::isVersionLower

### PR DESCRIPTION
**Pull request by luc-guyot-infomaniak**
The proposed changes replace a custom logic by a standard implementation of the lexicographical comparison of `std::vector<int>`.

### CODE REVIEW

File: src/libcommon/utility/utility.cpp [hunk 1]
```diff
@@ -728,19 +728,11 @@ bool CommonUtility::isVersionLower(const std::string &currentVersion, const std:
     extractIntFromStrVersion(targetVersion, targetTabVersion);
 
     if (currTabVersion.size() != targetTabVersion.size()) {
-        // Should not happen
```
Comment (luc-guyot-infomaniak) [inline src/libcommon/utility/utility.cpp:731]: The main changes.

File: src/libcommon/utility/utility.cpp [hunk 2]
```diff
@@ -728,19 +728,11 @@ bool CommonUtility::isVersionLower(const std::string &currentVersion, const std:
...
-            return false;
-        } else if (currTabVersion[i] < targetTabVersion[i]) {
-            return true;
-        }
+        return false; // Should not happen
     }
 
-    return false;
+    return std::lexicographical_compare(currTabVersion.cbegin(), currTabVersion.cend(), targetTabVersion.cbegin(),
+                                        targetTabVersion.cend());
```
Comment (herve-er) [inline src/libcommon/utility/utility.cpp:735]:
```suggestion
    return std::ranges::lexicographical_compare(currTabVersion, targetTabVersion);
```

**Comment by herve-er** [comment]
Well done! 
Additionally, it could be interesting to add some unit tests for the double-digit version (even if we do not use them for now).

```
//Double digit
CPPUNIT_ASSERT(CommonUtility::isVersionLower("1.0.0", "55.0.0"));
CPPUNIT_ASSERT(CommonUtility::isVersionLower("9.0.0", "55.0.0"));

CPPUNIT_ASSERT(CommonUtility::isVersionLower("25.0.0", "55.0.0"));
CPPUNIT_ASSERT(!CommonUtility::isVersionLower("75.0.0", "55.0.0"));

CPPUNIT_ASSERT(CommonUtility::isVersionLower("55.0.25", "55.0.55"));
CPPUNIT_ASSERT(!CommonUtility::isVersionLower("55.0.75", "55.0.55"));

CPPUNIT_ASSERT(CommonUtility::isVersionLower("55.25.0", "55.55.0"));
CPPUNIT_ASSERT(!CommonUtility::isVersionLower("55.75.0", "55.55.0"))
```

File: src/libcommon/utility/utility.cpp (hunk 2 above)
Comment (luc-guyot-infomaniak) [reply src/libcommon/utility/utility.cpp:735]: Fantastic! Changed to the concise call in the next commit.

Comment (luc-guyot-infomaniak) [reply src/libcommon/utility/utility.cpp]: Done.

**Comment by luc-guyot-infomaniak** [comment]
> Well done! Additionally, it could be interesting to add some unit tests for the double-digit version (even if we do not use them for now).
> 
> ```
> //Double digit
> CPPUNIT_ASSERT(CommonUtility::isVersionLower("1.0.0", "55.0.0"));
> CPPUNIT_ASSERT(CommonUtility::isVersionLower("9.0.0", "55.0.0"));
> 
> CPPUNIT_ASSERT(CommonUtility::isVersionLower("25.0.0", "55.0.0"));
> CPPUNIT_ASSERT(!CommonUtility::isVersionLower("75.0.0", "55.0.0"));
> 
> CPPUNIT_ASSERT(CommonUtility::isVersionLower("55.0.25", "55.0.55"));
> CPPUNIT_ASSERT(!CommonUtility::isVersionLower("55.0.75", "55.0.55"));
> 
> CPPUNIT_ASSERT(CommonUtility::isVersionLower("55.25.0", "55.55.0"));
> CPPUNIT_ASSERT(!CommonUtility::isVersionLower("55.75.0", "55.55.0"))
> ```

Thanks a lot! I added them in my last commit.

**Review by herve-er — APPROVED** [review]

**Pull request merged**